client/lib/desktop/index.js

```diff
@@ -204,7 +205,7 @@ const Desktop = {
 		debug( 'Sending logged-in = ' + status );
 
 		ipc.send( 'user-login-status', status );
-		ipc.send( 'user-auth', user(), oAuthToken.getToken() );
+		return ipc.invoke( 'user-auth', user(), oAuthToken.getToken() );
```

Comment:
Context: the `client/lib/desktop/index.js` `init()` method is invoked by Calypso's boot process in `client/boot/common.js` ([link](https://github.com/Automattic/wp-calypso/blob/91d5605b0ea4211a5752c26373888aec90233cef/client/boot/common.js#L384)):

```
if ( config.isEnabled( 'desktop' ) ) {
		require( 'lib/desktop' ).default.init();
}
```

Electron's `ipc.invoke` API returns a promise that is resolved when the main process responds to the invoke call ([Electron docs](https://www.electronjs.org/docs/api/ipc-main#ipcmainhandlechannel-listener)).

**Problem**: despite this being apparently sound in theory, this still doesn't solve the problem. When the application is booted, the `admin-bar-hours-scale` graphic is fails to load with a `403 forbidden` status, indicating some sort of authentication issue. A Developer Console reload fixes the graphic. In fact, logging to the developer console indicates that the request for the `admin-bar-hours-scale` asset is _still_ made prior to this init method returning:

<p align="center">
<img width="350" alt="asset-request-init-bug" src="https://user-images.githubusercontent.com/8979548/96898417-ae141d80-145d-11eb-9306-bdacc1809bfc.png">
</p>

The problem appears to be some strange timing-of-events issue that occurs only when a new application window is being created.

Reply:
Update: based on a suggestion by @belcherj, I refactored to block the render on the user authentication step in the main process. This appears to be working, and e2e tests are passing. See https://github.com/Automattic/wp-calypso/pull/46621/commits/0771f880f4f6e8e652a1981d266529da86f04d9b.

Still a little nervous as hopefully nothing else in the app is broken as a result - will be testing this patch a little more.